app/lib/relay/withRelayOptions.tsx

```diff
@@ -15,17 +16,32 @@ const withRelayOptions: WiredOptions<any> = {
     return createServerEnvironment({ cookieHeader: ctx?.req?.headers.cookie });
   },
   serverSideProps: async (ctx: NextPageContext) => {
+    const { default: getAuthRole } = await import('../../utils/getAuthRole');
     // Server-side redirection of the user to their landing route, if they are logged in
     const request = ctx.req as any;
+    const authRole = getAuthRole(request);
     const authenticated = isAuthenticated(request);
-    // They're logged in.
-    if (authenticated) {
+    const routeAuthorized = isRouteAuthorized(request.url, authRole.pgRole);
+    // They're logged in and authorized to access the page or the page is not protected
+    if (routeAuthorized) {
       return {};
     }
-    // Handle not logged in
+
+    // They're logged in but they are not authorized to access the page
+    if (authenticated) {
+      return {
+        redirect: {
+          destination: authRole.landingRoute,
+        },
+      };
+    }
+
+    // Redirect them to landing route responding to the page they are trying to access
     return {
       redirect: {
-        destination: '/',
+        destination: request.url.startsWith('/analystportal')
+          ? '/analystportal'
```

Comment:
remember to replace all the `/analystportal` to `/analyst`, as per the issue's acceptance criteria